shared/src/hooks/useFetchVaultData.ts

```diff
@@ -111,9 +111,11 @@ const useFetchVaultData = (
             {
               ...prev.responses[vault],
               ...response,
-              vaultMaxWithdrawAmount: vaultMaxWithdrawableShares
-                .mul(response.deposits)
-                .div(totalSupply),
+              vaultMaxWithdrawAmount: !totalSupply.isZero
+                ? vaultMaxWithdrawableShares
+                    .mul(response.deposits)
+                    .div(totalSupply)
+                : BigNumber.from(0),
```

Comment:
I don't think the math is correct here since it doesn't take into account pending deposits and withdrawals. A better approach to this is to fetch the price per share and compute it.

```js
vaultMaxWithdrawableShares.mul(pricePerShare).div(10**decimals)
```

Reply:
This one is v1 vault, and since we can get vaultMaxWithdraw in share from contract directly, we merely need to turn it into asset amount. The one you describe are for v2 I believe